### PR TITLE
add support to change connected body for joint component

### PR DESCRIPTION
**Pull request by yiwenxue**
Re: #https://github.com/cocos/cocos-engine/issues/14366

### Changelog

* add support to change connected body for joint component

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/physics/physx/joints/PhysXJoint.cpp [hunk 1]
```diff
@@ -69,16 +69,32 @@ void PhysXJoint::setConnectedBody(uint32_t rigidBodyID) {
...
         _mJoint->setActors(_mSharedBody->getImpl().rigidActor, _mConnectedBody ? _mConnectedBody->getImpl().rigidActor : nullptr);
     }
+
+    if (oldConnectedBody) {
+        if (oldConnectedBody->isDynamic()) {
+            oldConnectedBody->getImpl().rigidDynamic->wakeUp();
+        }
+    }
+
+    updateScale0();
```
Comment (lealzhan) [inline native/cocos/physics/physx/joints/PhysXJoint.cpp:96]: updateScale0() and updateScale1() should have more specific name

File: cocos/physics/physx/joints/physx-joint.ts [hunk 2]
```diff
@@ -40,7 +40,33 @@ export class PhysXJoint implements IBaseConstraint {
...
+        if (v) {
+            const sb2 = (v.body as PhysXRigidBody).sharedBody;
+            setJointActors(this._impl, sb.impl, sb2.impl);
+            sb2.addJoint(this, 1); // add to new sb2
+        } else {
+            setJointActors(this._impl, sb.impl, null);
+        }
+
+        if (oldBody2) {
+            oldBody2.wakeUp(); // wake it up, or the old body will be sleep for a while
```
Comment (lealzhan) [inline cocos/physics/physx/joints/physx-joint.ts:64]: sleeping?

Comment (yiwenxue) [reply cocos/physics/physx/joints/physx-joint.ts:64]: Yes, it's possible, especially when the input rigid body has just been created.

File: native/cocos/physics/physx/joints/PhysXJoint.cpp (hunk 1 above)
Comment (yiwenxue) [reply native/cocos/physics/physx/joints/PhysXJoint.cpp:96]: `updateScale0` and `updateScale1` are inherited from their parents. We could consider refactoring them in the future.

**Review by lealzhan — APPROVED** [review]

**Pull request merged**